src/main/java/hudson/plugins/ec2/InstanceInfo.java

```diff
@@ -0,0 +1,29 @@
+package hudson.plugins.ec2;
+
+import com.amazonaws.services.ec2.model.Instance;
+import com.amazonaws.services.ec2.model.KeyPair;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class InstanceInfo {
+    private Instance instance;
+    private KeyPair keypair;
```

Comment:
```suggestion
    private final Instance instance;
    private final KeyPair keypair;
```
(Candidate for `record` when on 2.479+ and thus Java 17.)